source/opt/passes.h

```diff
@@ -74,6 +80,165 @@ class EliminateDeadConstantPass : public Pass {
   bool Process(ir::Module*) override;
 };
 
+// The optimization pass that folds "Spec Constants", which are defined by
+// OpSpecConstantOp and OpSpecConstantComposite instruction, to "Normal
+// Constants", which are defined by OpConstantTrue, OpConstantFalse,
+// OpConstant, OpConstantNull and OpConstantComposite instructions. A Spec
+// Constant is foldable if all of its value(s) can be determined from the
+// module. For example, an Integer Spec Constant can be folded if its value
```

Comment:
"For example ..." sentence is a bit vague.  It's not clear if it refers to OpSpecConstant with an integer type, or whether it refers to an OpSpecConstantOp that results in an integer type.  It should only apply to the second case.